### ISSUE TITLE
Process hangs and won't exit correctly when sending incorrect message from DEALER->REP

### ISSUE THREAD

**Issue by Madd0g** [zeromq/zeromq.js#239]
I have the same DEALER code sending messages to REP/ROUTER, the only difference that sending to REP requires an extra `.send()` in front of the actual message.

```
let socket = zmq.socket('dealer');
socket.bindSync('tcp://127.0.0.1:5555');
....
if (isRep) {
    socket.send('', zmq.ZMQ_SNDMORE);
}
socket.send(envelope);
```

Right now if I forget to pass `isRep` the message is not received, but the process also hangs after CTRL+C and I have to kill it with `kill -9`.

Anything I can do to catch this error and prevent it? This seems like such an easy mistake to make, I keep forgetting to pass the isRep.

Thanks

**Comment by rolftimmermans** [comment]
Recently we have released 6.0 beta. It features a [new API](https://github.com/zeromq/zeromq.js#examples) that addresses some [fundamental issues](https://github.com/zeromq/zeromq.js/issues/189) with the previous API and also addresses a number of stability bugs. To make upgrading easier it includes a [compatibility layer](https://github.com/zeromq/zeromq.js#compatibility-layer-for-version-45) with versions 4.x/5.x. It would be great if you could give the latest version a spin to see if this solves this particular issue. If you run into any problems with it, feel free to report it here or in a new issue.

**Comment by aminya** [comment]
v6 was released. Please try again with the latest version, and report back if the issue still persists.
https://github.com/zeromq/zeromq.js/releases/tag/v6.0.0
